command/harness/dlite/dlite.go

```diff
@@ -171,3 +181,52 @@ func (c *dliteCommand) run(*kingpin.ParseContext) error {
 	}
 	return nil
 }
+
+func (c *dliteCommand) setupPool(ctx context.Context) (*config.PoolFile, error) {
+	instanceStore, stageOwnerStore, err := database.ProvideStore(c.env.Database.Driver, c.env.Database.Datasource)
+	if err != nil {
+		logrus.WithError(err).Fatalln("Unable to start the database")
+	}
+	c.poolManager = drivers.NewManager(ctx, instanceStore, stageOwnerStore, &c.env)
+	poolConfig, err := harness.SetupPool(ctx, &c.env, c.poolManager, c.poolFile)
+	if err != nil {
+		logrus.WithError(err).Error("could not setup pool")
+		return poolConfig, err
+	}
+	c.metrics.AddMetricStore(&metric.Store{
+		Store:       instanceStore,
+		Query:       nil,
+		Distributed: false,
+	})
+	return poolConfig, nil
+}
+
+func (c *dliteCommand) setupDistributedPool(ctx context.Context) (*config.PoolFile, error) {
+	logrus.Infoln("Starting postgres database")
+	instanceStore, stageOwnerStore, err := database.ProvideStore(c.env.Database.Postgres.Driver, c.env.Database.Postgres.Datasource)
+	if err != nil {
+		logrus.WithError(err).Fatalln("Unable to start the database")
+		return nil, err
+	}
+	c.distributedPoolManager = drivers.NewDistributedManager(drivers.NewManager(ctx, instanceStore, stageOwnerStore, &c.env))
+	poolConfig, err := harness.SetupPool(ctx, &c.env, c.distributedPoolManager, c.poolFile)
+	if err != nil {
+		logrus.WithError(err).Error("could not setup distributed pool")
+		return poolConfig, err
+	}
+	c.metrics.AddMetricStore(&metric.Store{
+		Store: instanceStore,
+		Query: &types.QueryParams{
+			RunnerName: c.env.Runner.Name,
+		},
+		Distributed: true,
+	})
+	return poolConfig, nil
+}
+
+func (c *dliteCommand) getPoolManager(distributed bool) drivers.IManager {
```

Comment:
Can we just use a single pool manager `c.poolManager` and set it up once in the beginning. This would also help once we do dependency injection and is a cleaner approach.